Scala 3 error reporting crash for bad `summonInline`

**Describe the bug**

Scala 3 (v3.0.0-RC3) `summonInline` crashes the metals error reporting.

**To Reproduce**

Use the following code:
```scala
inline def mark[T] = compiletime.summonInline[T]
val x = mark[1 =:= 2]
```

**Expected behavior**

Error should be reported for `x`

**Installation:**

- Operating system: Windows
- Editor: Visual Studio Code
- Metals version: v0.10.2

**Additional context**

Crash log
```
��� 08, 2021 4:00:54 ������ org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleNotification
WARNING: Notification threw an exception: {
  "jsonrpc": "2.0",
  "method": "build/publishDiagnostics",
  "params": {
    "textDocument": {
      "uri": "file:///c:/IdeaProjects/dtest/library/src/scala/compiletime/package.scala"
    },
    "buildTarget": {
      "uri": "file:/C:/IdeaProjects/dtest/?id\u003droot"
    },
    "diagnostics": [
      {
        "range": {
          "start": {
            "line": 0,
            "character": 0
          },
          "end": {
            "line": 0,
            "character": 0
          }
        },
        "severity": 1,
        "message": "cannot reduce summonFrom with\n patterns :  case \u001b[33mgiven \u001b[0mt @ _:\u001b[35m(\u001b[31m1\u001b[0m : Int) \u003d:\u003d (\u001b[31m2\u001b[0m : Int)\u001b[0m"
      }
    ],
    "reset": true
  }
}
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: java.nio.file.NoSuchFileException: c:\IdeaProjects\dtest\library\src\scala\compiletime\package.scala
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:235)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:370)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:421)
	at java.base/java.nio.file.Files.readAllBytes(Files.java:3205)
	at scala.meta.internal.io.PlatformFileIO$.slurp(PlatformFileIO.scala:45)
	at scala.meta.internal.io.FileIO$.slurp(FileIO.scala:24)
	at scala.meta.internal.mtags.CommonMtagsEnrichments$XtensionAbsolutePath.toInput(CommonMtagsEnrichments.scala:435)
	at scala.meta.internal.metals.Diagnostics.onPublishDiagnostics(Diagnostics.scala:143)
	at scala.meta.internal.metals.Diagnostics.onBuildPublishDiagnostics(Diagnostics.scala:123)
	at scala.meta.internal.metals.ForwardingMetalsBuildClient.onBuildPublishDiagnostics(ForwardingMetalsBuildClient.scala:99)
	... 15 more
```
**Search terms**

summon

--- a/mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala
+++ b/mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala
@@ -40,6 +40,13 @@ object SemVer {
         .exists { case (a, b) => a > b }
     }
 
+    override def toString: String =
+      List(
+        Some(s"$major.$minor.$patch"),
+        releaseCandidate.map(s => s"-RC$s"),
+        milestone.map(s => s"-M$s")
+      ).flatten.mkString("")
+
   }
 
   object Version {

--- a/tests/slow/src/test/scala/tests/feature/SyntaxErrorLspSuite.scala
+++ b/tests/slow/src/test/scala/tests/feature/SyntaxErrorLspSuite.scala
@@ -3,6 +3,7 @@ package tests.feature
 import scala.concurrent.Future
 
 import scala.meta.internal.metals.{BuildInfo => V}
+import scala.meta.internal.semver.SemVer
 
 import munit.Location
 import tests.BaseLspSuite
@@ -332,6 +333,36 @@ class SyntaxErrorLspSuite extends BaseLspSuite("syntax-error") {
       )
       _ <- server.didOpen("a/src/main/scala/A.scala")
       _ = assertEmpty(client.workspaceDiagnostics)
+    } yield ()
+  }
+
+  test("scala3-inline-position") {
+    cleanWorkspace()
+    // works from 3.0.1-RC1
+    val latestScala3 =
+      V.nonDeprecatedScalaVersions
+        .map(SemVer.Version.fromString)
+        .filter(_.major == 3)
+        .sortWith(_ > _)
+        .head
+    for {
+      _ <- server.initialize(
+        s"""
+           |/metals.json
+           |{"a": { "scalaVersion": "$latestScala3" }}
+           |/a/src/main/scala/A.scala
+           |inline def mark[T] = compiletime.summonInline[T]
+           |val x = mark[1 =:= 2]
+           |""".stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/A.scala")
+      _ = assertNoDiff(
+        client.workspaceDiagnostics,
+        """|a/src/main/scala/A.scala:2:16: error: Cannot prove that (1 : Int) =:= (2 : Int).
+           |val x = mark[1 =:= 2]
+           |               ^^^
+           |""".stripMargin
+      )
     } yield ()
   }
 

